src/libraries/System.Runtime/tests/System.Runtime.Tests/System/HalfTests.cs

```diff
@@ -400,6 +400,14 @@ public static IEnumerable<object[]> ExplicitConversion_ToSingle_TestData()
                 (BitConverter.UInt16BitsToHalf(0b1_00001_0000000000), BitConverter.Int32BitsToSingle(unchecked((int)0xB8800000))) // highest negative normal
             };
 
+            if (PlatformDetection.IsRiscV64Process) // RISC-V does not preserve payload
+            {
+                data[12] = (BitConverter.UInt16BitsToHalf(0b0_11111_1010101010),
```

Comment:
It is hard to match this with the 12th and 13th test cases above. 

Instead of modifying the test cases, can we relax the logic that compares expected and actual for RISC-V?

Reply:
Do you mean something like changes from this [commit](https://github.com/dotnet/runtime/pull/96888/commits/9355eea055d8743424aab1f59be0ac78fa4acdae)? 
Or maybe just to skip NaN cases for RISC-V due to specific behavior for NaN input? (skip assert)

Reply:
I think Jan is saying it'd be nicer if we add an `AssertExtensions.Equal(float expected, float actual)` method which handles this, so we can just switch out the assert and not the rest of the logic.

We actually have helper that would work already which doesn't check NaN payload at all: https://github.com/dotnet/runtime/blob/afb0d69fcdd11daffce9f50e90daf601ec026df1/src/libraries/Common/tests/TestUtilities/System/AssertExtensions.cs#L824. This exists to allow a small amount of variance in results due to differences in implementations for the Math functions which defer to the C runtime (libm).

So, we could either call that and no longer check payload preservation on any platform, or we could add a new helper that works like the following:
```csharp
public static void Equal(float expected, float actual)
{
    if (BitConverter.SingleToInt32Bits(expected) == BitConverter.SingleToInt32Bits(actual))
    {
        return;
    }

    if (PlatformDetection.IsRiscV64Process && float.IsNaN(expected) && float.IsNaN(actual))
    {
        // RISC-V does not preserve payload
        return;
    }

    throw EqualException.ForMismatchedValues(ToStringPadded(expected), ToStringPadded(actual));
}
```

I think adding a new helper is a little better for now and if we decide to not preserve payload anywhere, it becomes a 1 line change to account for that (`ToStringPadded` is currently a local function in the existing helper and would need to be extracted).

Reply:
It's done I guess, could you review?